src/Ports/Symfony/SymfonyPublisher.php

```diff
@@ -34,7 +35,50 @@ public function __construct(EventDispatcherInterface $dispatcher)
      */
     public function publish(DomainEvent $event): void
     {
-        $this->dispatcher->dispatch(\get_class($event), new EventAdapter($event));
+        if ($this->dispatcher instanceof ContractInteface) {
```

Comment:
not sure how i feel about this, im still torn about its none dependency-dependency on the symfony libraries of 2 different versions. there is some kind of code smell with this library from that. is there a different pattern that could be used that can have the dependency of symfony separate, or support by the different tag version (v2 - sf4, v3 - sf5) or not have the dependency at all?

for forward compatibility, wouldn't the most "current" be the default.
```
if (EventDispatcherInterfaceComponent) {
 // call dispatch function
return
}

//new contract here
```

Reply:
The API for the event dispatcher changed from version <=4.4 to 5.0. In order to keep BC with users that uses this package, I had to make sure both versions are still usable.
The API of the publisher did not changed, so I don't want users to have errors on my package when only updating the event dispatcher to 5.0. I agree its not the best solution, but it the one I found to keep BC.

Reply:
In a future release, we could remove this dependancy, and increase the requirement to a more recent one.

Reply:
@mkoert I created #18 to review this case. thanks